drivers/serial/uart_stm32.c

```diff
@@ -815,7 +815,7 @@ static void uart_stm32_irq_config_func_##index(const struct device *dev)	\
 STM32_UART_IRQ_HANDLER_DECL(index);					\
 									\
 static const struct soc_gpio_pinctrl uart_pins_##index[] =		\
-					ST_STM32_DT_PINCTRL(0, index);	\
+				ST_STM32_DT_INST_PINCTRL(0, index);	\
```

Comment:
`DT_INST*` macros take instance index as a first argument, so I think we should do the same to keep consistency.

Reply:
Ok, just added a new commit to change this as well.